### ISSUE TITLE
Jetengine: Adding a a multi select option in the new front end forms

### ISSUE THREAD

**Issue by asmarove** [Crocoblock/suggestions#878]
There is no option to have a multi select for (select, radio buttons, Checkboxes) in the new front end forms
, while it's available in the normal custom post type.
![2](https://user-images.githubusercontent.com/32307693/73485567-fd845c00-43ab-11ea-93d1-ed4603b69e03.png)
![1](https://user-images.githubusercontent.com/32307693/73485568-fd845c00-43ab-11ea-9c52-29efc0497369.png)

**Comment by arekuki** [comment]
+1

**Comment by ahmadrusli** [comment]
Yes. please add to new version. we need to select some countries. Thank you :)

**Comment by designboxmedia** [comment]
Yes please add this. We are using check boxes when they should be multiple select.  
Chosen select would be a bonus too. https://harvesthq.github.io/chosen/

**Comment by anderoav** [comment]
Any news?

**Comment by stasonua0** [comment]
+1

**Comment by imhasem** [comment]
is it possible to add select filed with search?

**Comment by moxet** [comment]
+1

**Comment by Roberto77** [comment]
+1

**Comment by iluvios** [comment]
Feature still missing :(

**Comment by Nelicquele** [comment]
+1 I really need this feature. 

I've got a few requirements for potential clients before they can sign up. I have to be able te require to select each box before they can continue.

**Comment by Nelicquele** [comment]
+1 I really need this option!

**Comment by yuleidyvdres** [comment]
+1 This feature will be really helpful

**Comment by mianrashid** [comment]
damn this feature should be there and this just stopped me from buying it.. I hope this feature was there with chosen js... saved me 750$

**Comment by nreljed** [comment]
#3885

**Comment by claujohnson** [comment]
Unfortunately this solution is not yet complete in Crocoblock :(

**Comment by developernerdsacademy** [comment]
We REALLY need this feature!!! I have thousands of posts and when users add related posts, the form is so long! I would rather have a multi select for that or at least a way to make the checkbox field have a scroll instead of being a mile long!

**Comment by iluvios** [comment]
Yepp, still need this feature

**Comment by nreljed** [comment]
this function can only come with select2.
unite your ranks and voices, in this post #3885
and tag all the possible problems linked to the select ecosystem
as the dev forces are now divided between jetformbuilder and jetengine forms, this will become even more complicated to manage soon...
at some point, a choice should be made, and it will be the right one :)
be patient

**Comment by Joseph089** [comment]
+1

**Comment by honzabit** [comment]
Hey, I created a small plugin that adds a new `Select2` field component. The plugin works for _single_ or _multiple_ values (configurable). It's still an early release and I'm not sure yet it supports all the field options of the normal select component yet (can't find a way so far to solve this without duplicating code). It also works nicely inside repeaters as it observes newly added select elements and transforms them on the fly.

Anyway, you can grab it from [here](https://www.buymeacoffee.com/ian.d) and let me know if something breaks so that I can fix it.

P.S: you obviously don't need to buy me a coffee and the code is under GPLv3

**Comment by ghost** [comment]
> here

Hello, I have used it but it didn't work. I couldn't fill the options with custom taxonomy terms... anyway it's nice to see somebody trying to fix this lack of feature.

**Comment by honzabit** [comment]
@lucasvlc makes sense. That's why I said above that `I'm not sure yet it supports all the field options of the normal select component yet (can't find a way so far to solve this without duplicating code)`. The problem remains though and is not really the duplication that I care about, but the fact that it can break if crocoblock devs decide to alter the way they populate these field options. In any case, I just published a new version of the plugin, feel free to give it a shot and let me know.

**Comment by newtonata** [comment]
+1

**Comment by aCodeDragon** [comment]
+1million ;)

**Comment by morosmo** [comment]
+1

**Comment by stefanradu9** [comment]
> Hey, I created a small plugin that adds a new `Select2` field component. The plugin works for _single_ or _multiple_ values (configurable). It's still an early release and I'm not sure yet it supports all the field options of the normal select component yet (can't find a way so far to solve this without duplicating code). It also works nicely inside repeaters as it observes newly added select elements and transforms them on the fly.
> 
> Anyway, you can grab it from [here](https://www.buymeacoffee.com/ian.d) and let me know if something breaks so that I can fix it.
> 
> P.S: you obviously don't need to buy me a coffee and the code is under GPLv3

Amazing work, thank you!!

**Comment by stapolin** [comment]
This is something I would love to see added too. I build directory websites, and being able to choose multiple categories/taxonomies in a select dropdown while adding or editing a listing from the frontend would be amazing.

**Comment by Ruudolfs** [comment]
+1

**Comment by stapolin** [comment]
Just to add to my last comment.. 

Being able to choose a maximum number of items would be handy.. especially for directory sites where you may only allow a listing in up to 3 categories.

**Comment by satomiao** [comment]
+1

**Comment by menzer2015** [comment]
Multiselect please +1 (2 years on) - lol

**Comment by fbeach73** [comment]
+1 this is very necessary, and holding a lot of things up in the air for front end forms and options on taxonomy buildouts

**Comment by menzer2015** [comment]
Come on guys multi select is a powerful tool it also helps with reducing layout clutter, I use chioces.js to convert multiple select boxes so you get a neat select dropdown but as you don't even have the multi-select as an option is causing me to use gravityforms but I would prefer to use jetformbuilder. please please release this after 2 years request

**Comment by menzer2015** [comment]
BTW you only need this jquery to convert your select box to a dropdown box (Make sure to change the ID to your targeted ID)
  $("#points_of_interest").attr("multiple",''); <--- adds multiple to the selectbox

If you want it to look nice like the demo below then I have included a little snippet that may help some people, if you have a better solution please let me know but I hope this can help someone

Install the choices jquery and enqueue the jquery script
Demo and code here:
https://choices-js.github.io/Choices/

After you have enqueued the above script initiate the script with this code. Add this to your function file or snippet plugin make sure to change my ID below

```
add_action('wp_footer', 'jw_choices_select');
function jw_choices_select(){?>
<script>
jQuery(document).ready(function( $ ) {
  //Place your jQuery code here...
  //// Pass jQuery element
  $("#change_this_ID").attr("multiple",''); <---- this converts the simple select box to a dropdown multiple box which the choices.js jquery will convert into a beautiful select boxes
//  $('#change_this_ID').attr('data-calculate','25'); <-- I am still working on adding the data-calculate to the option of the select box for those that need to have calculations (I don't)

  var firstElement = document.getElementById('points_of_interest');
//	var firstElement = document.getElementsByClassName('checkradio-wrap');
var choices1 = new Choices(firstElement, {
    delimiter: ',',
    editItems: false,
    removeButton: true,
	searchEnabled: false,
	searchChoices: false,
	removeItemButton: false
});
  
});
</script>
<?php
}
```

**Comment by zivoriol** [comment]
+1000
Is there any update on this?

**Comment by ghost** [comment]
+1 Yes please!

**Comment by tusharttyagi** [comment]
+10000000, please. need this badly

**Comment by WordpressTodayAgency** [comment]
Just an update: They added Select2 with the option to select multiple values in JetFormBuilder and it's called Select Autocomplete https://jetformbuilder.com/addons/select-autocomplete/

**Comment by tusharttyagi** [comment]
> Just an update: They added Select2 with the option to select multiple values in JetFormBuilder and it's called Select Autocomplete https://jetformbuilder.com/addons/select-autocomplete/

THANK YOU LIFESAVER

**Comment by Perica888** [comment]
> Hey, I created a small plugin that adds a new `Select2` field component. The plugin works for _single_ or _multiple_ values (configurable). It's still an early release and I'm not sure yet it supports all the field options of the normal select component yet (can't find a way so far to solve this without duplicating code). It also works nicely inside repeaters as it observes newly added select elements and transforms them on the fly.
> 
> Anyway, you can grab it from [here](https://www.buymeacoffee.com/ian.d) and let me know if something breaks so that I can fix it.
> 
> P.S: you obviously don't need to buy me a coffee and the code is under GPLv3

Do you have an update of your cool add-on because its stopped working on part of multiple selection

**Comment by marrisonlab** [comment]
You can achieve it checkbox filter "dressed" like a select

![image](https://github.com/user-attachments/assets/8f6c0025-8edf-4dbe-9a13-2f1f1e51235e)

**Comment by Crocoblock** [comment]
This option is available in JetFormBuilder https://i.imgur.com/Htiya5J.png, please use it instead of JetEngine Forms
